test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1667,10 +1672,35 @@ def test_multiple_job_preemption_order(self):
 
         # Only One equivalence class for jid2 and jid3 is present since both
         # suspended jobs are of same type and running on same vnode
-        t = int(time.time())
+
         self.scheduler.log_match("Number of job equivalence classes: 2",
                                  max_attempts=10, starttime=self.t)
 
+        # Add a job to Queue state
+        a = {'Resource_List.ncpus': 1}
+        J = Job(TEST_USER, attrs=a)
+        jid6 = self.server.submit(J)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid6)
+
+        # Set scheduling = false
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'False'})
+
+        # Delete one running, one suspended job and one of high priority job
+        # This will leave 2 free nodes to pick up the suspended and queued job
+
+        self.server.deljob(jid1)
+        self.server.deljob(jid2)
+        self.server.deljob(jid5)
```

Comment:
Do you need to do wait=True?  If you don't, the jobs can take some time to actually be deleted.

Reply:
If wait is set to true, PTL starts a new scheduling cycle. Since we do not want this, we have not set wait.

Reply:
It isn't PTL that's running the cycle, it's PBS.  What is going to happen is that either the jobs exit before you run your cycle or they don't.  If they exit, then the cycle you start will run the jobs.  If the jobs haven't exited, then the cycle you run won't run the jobs.  When the jobs truly exit, a cycle will start and the jobs will run.  Why are you trying to avoid the cycle when jobs exit?

Reply:
@bhroam , when set wait=true it was PTL who was initiating scheduling =true till the job is dequeued.
We wanted suspend and queued jobs to be picked up in the same scheduling cycle by scheduler and make sure they are running hence we disabled the scheduling cycle.
when you say "When the jobs truly exit, a cycle will start and the jobs will run", do you mean it will kick the scheduling cycle even if the scheduler is off.

Reply:
@anamikau I looked through the deljob() code and didn't see it kicking a cycle.  Did you mean that the test itself is kicking a cycle?  I see that.  The problem with the test is that once the deljob() is issued, it might take several seconds for the job to finish and exit the system.  The resources won't be available until they do.  The cycle that the test kicks won't necessarily have the resources it requires to run the jobs.  I think the way to do what the test wants to do is to turn scheduling off, deljob() with wait=true for all 3 jobs, and then turn scheduling=true again.

PBS will not run any cycles if scheduling=False.  Sorry if I misled you.

Reply:
Also if you want you can call deljob with all jids like self.server.deljob([jid1, jid2, jid3])